src/Microsoft.Diagnostics.DebugServices.Implementation/ContextService.cs

```diff
@@ -26,8 +26,21 @@ public ContextService(IHost host)
 
             // Clear the current context when a target is flushed or destroyed
             host.OnTargetCreate.Register((target) => {
-                target.OnFlushEvent.Register(() => ClearCurrentTarget(target));
-                target.OnDestroyEvent.Register(() => ClearCurrentTarget(target));
+                target.OnFlushEvent.Register(() => {
+                    if (IsTargetEqual(target, _currentTarget))
```

Comment:
Why do we no longer want to set the `_currentTarget` to null upon flushing?

Reply:
The idea is not to lose the current target during a flush (especially if multiple targets). The target itself is flushed.